src/use_case/generate_routine/GenerateRoutineOutputData.java

```diff
@@ -6,12 +6,31 @@
 
 public class GenerateRoutineOutputData {
     private ArrayList<Exercise> listOfExercises;
+    private String name;
+    private int sets;
+    private int reps;
 
-    public GenerateRoutineOutputData(ArrayList<Exercise> listOfExercises) {
+    public GenerateRoutineOutputData(ArrayList<Exercise> listOfExercises, String name, int sets, int reps) {
```

Comment:
Here, I don't think your output data requires sets or reps: all you need is name and listOfExercise. Because your list of exercises will already contain all the sets and reps for each exercise inside each exercise entity. int sets and int reps here doesn't really make sense; sets and reps for the whole routine? Unless you're using those two values later—I don't know your use case perfectly, but if not, I think this needs to be removed.